pkg/cli/login/loginoptions.go

```diff
@@ -124,13 +124,18 @@ func (o *LoginOptions) getClientConfig() (*restclient.Config, error) {
 	o.Server = serverNormalized
 	clientConfig.Host = o.Server
 
-	// use specified CA or find existing CA
-	if len(o.CAFile) > 0 {
-		clientConfig.CAFile = o.CAFile
-		clientConfig.CAData = nil
-	} else if caFile, caData, ok := findExistingClientCA(clientConfig.Host, *o.StartingKubeConfig); ok {
-		clientConfig.CAFile = caFile
-		clientConfig.CAData = caData
+	if !o.InsecureTLS {
+		// use specified CA or find existing CA
+		if len(o.CAFile) > 0 {
+			clientConfig.CAFile = o.CAFile
+			clientConfig.CAData = nil
+		} else if caFile, caData, ok := findExistingClientCA(clientConfig.Host, *o.StartingKubeConfig); ok {
+			clientConfig.CAFile = caFile
+			clientConfig.CAData = caData
+		}
+	} else {
+		// if user has selected option --insecure-skip-tls-verify=true, TLS server certificate verification should not be attempted
+		clientConfig.Insecure = true
```

Comment:
this alone should work, no? If not, explain in another comment why you specifically leave the above CA data unpopulated

Reply:
Previous [attempt ](https://github.com/openshift/oc/pull/746) was exactly doing this but it failed in here https://github.com/openshift/oc/blob/f7835631278b03e4388f65145b56e1fd93bf4e37/vendor/k8s.io/client-go/transport/transport.go#L64-L66.

I think client-go transport wants CA fields are empty if `insecure` is set.